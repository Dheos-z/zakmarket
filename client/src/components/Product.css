.cardFooter {
    display: flex;
    justify-content: space-around;
}

.productDesign {
    background-color: rgb(232, 236, 228);
    text-align: center;
}