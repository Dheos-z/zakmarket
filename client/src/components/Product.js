import React, { Component } from 'react';
import { Card, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import './Product.css';



class Product extends Component {

    toFirstLetterUpperCase = str => {
        return str.charAt(0).toUpperCase() + str.slice(1);
    }


    get productDetails() {

        const { product } = this.props;
        const productKeys = Object.keys(product);
        const ignoreKeys = ['idProduct', 'owner', 'price', 'name', 'type'];

        return (
            <span className="productDetails">
                {
                    productKeys.map((key, index) => {
                        if (ignoreKeys.includes(key)) return (null);

                        return (
                            <span key={index}>
                                {this.toFirstLetterUpperCase(key) + ': ' + product[key]}
                                <br/>
                            </span>
                        )
                    })
                }
            </span>
        );
    }

    render() {
        const { product } = this.props;
        return (
            <Card>
                <Card.Header>{this.toFirstLetterUpperCase(product.type)}</Card.Header>
                <Card.Body>
                    <Card.Title>{this.toFirstLetterUpperCase(product.name)}</Card.Title>
                    <Card.Subtitle>{product.price}</Card.Subtitle>
                    <Card.Text>
                        {this.productDetails}
                    </Card.Text>
                </Card.Body>
                <Card.Footer>
                    <div className="cardFooter">
                        <div className="footerBtn">
                            <Link to={{ pathname: '/updateProduct', state: { product } }}>
                                <Button variant="secondary">Update</Button>
                            </Link>{' '}
                        </div>
                        <div className="footerBtn">
                            <Button onClick={() => this.props.handleDelete(product.idProduct)} variant="danger">Delete</Button>
                        </div>
                    </div>
                </Card.Footer>
            </Card>
        )
    }
}


export default Product;