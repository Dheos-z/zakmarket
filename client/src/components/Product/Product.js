// The product component is represented by 1 product displayed in the Products component

import React from 'react';
import { Card, Button } from 'react-bootstrap';
import { Link, BrowserRouter } from 'react-router-dom';
import './Product.css';
import history from '../../history';


const Product = props => {

    const toFirstLetterUpperCase = str => {
        return str.charAt(0).toUpperCase() + str.slice(1);
    }

    const productDetails = () => {
        const productKeys = Object.keys(props.product);
        const ignoreKeys = ['idProduct', 'owner', 'price', 'name', 'type'];

        return (
            <span className="productDetails">
                {
                    productKeys.map((key, index) => {
                        if (ignoreKeys.includes(key)) return (null);

                        return (
                            <span key={index}>
                                {toFirstLetterUpperCase(key) + ': ' + props.product[key]}
                                <br />
                            </span>
                        )
                    })
                }
            </span>
        );
    }

    return (
        <Card>
            <Card.Header className='productDesign'>{props.product.type.toUpperCase()}</Card.Header>
            <Card.Body className="cardBody">
                <Card.Title>{toFirstLetterUpperCase(props.product.name)}</Card.Title>
                <Card.Subtitle>{props.product.price + " $"}</Card.Subtitle>
                <Card.Text>
                    {productDetails()}
                </Card.Text>
            </Card.Body>
            <Card.Footer className='productDesign'>
                <div className="cardFooter">
                    <div className="footerBtn">
                        <BrowserRouter>
                            <Link
                                to={{ pathname: '/updateProduct', state: { product: props.product} }}
                                onClick={() => history.push('/updateProduct')}>
                                <Button variant="secondary">Update</Button>
                            </Link>
                        </BrowserRouter>
                    </div>
                    <div className="footerBtn">
                        <Button onClick={() => props.handleDelete(props.product.idProduct)} variant="danger">Delete</Button>
                    </div>
                </div>
            </Card.Footer>
        </Card>
    )
}


export default Product;