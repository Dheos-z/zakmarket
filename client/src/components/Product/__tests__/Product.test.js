import React from 'react';
import renderer from 'react-test-renderer';
import Product from '../Product';

describe('Product', () => {
  const product = {
    idProduct: 12,
    name: 'Test product',
    price: 22,
    details: 'Details of the test product',
    type: 'test'
  };

  const handleDelete = jest.fn();

  test('snapshot renders', () => {
    const component = renderer.create(<Product product={product} handleDelete={handleDelete}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
