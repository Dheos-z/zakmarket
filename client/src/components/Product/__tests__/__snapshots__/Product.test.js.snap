// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product snapshot renders 1`] = `
<div
  className="card"
>
  <div
    className="productDesign card-header"
  >
    TEST
  </div>
  <div
    className="cardBody card-body"
  >
    <div
      className="card-title h5"
    >
      Test product
    </div>
    <div
      className="card-subtitle h6"
    >
      22 $
    </div>
    <p
      className="card-text"
    >
      <span
        className="productDetails"
      >
        <span>
          Details: Details of the test product
          <br />
        </span>
      </span>
    </p>
  </div>
  <div
    className="productDesign card-footer"
  >
    <div
      className="cardFooter"
    >
      <div
        className="footerBtn"
      >
        <a
          href="/updateProduct"
          onClick={[Function]}
        >
          <button
            className="btn btn-secondary"
            disabled={false}
            type="button"
          >
            Update
          </button>
        </a>
      </div>
      <div
        className="footerBtn"
      >
        <button
          className="btn btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
